### PR TITLE
Feature - New API

### PR DESCRIPTION
**Pull request by jfo84**
This has been a mess @pierpo. Apologies

Changes
- ArcherElement `relations` API

The attribute was too open-ended, which necessitated checks when merging relations and manually merging from ID's. We instead generate the `SourceToTargetType` from the `RelationType`

The `fromTos` state is replaced internally with `sourceToTargets`. The API is essentially the same except from and to have been replaced with source and target

- Expose svgContainerStyle

This lets you change the z-index of the SVG's, which I found quite useful

- Add react-test-renderer and react-fast-compare

The former helps with testing snapshots in jest. The latter is a tiny library that adds some sugar to avoid the circular references when comparing stringified React elements

Overall, I think it makes the library more reliable, less complex, and faster. That's definitely it for changes

### CODE REVIEW

**Comment by pierpo** [comment]
Wow, that's quite a lot of changes 😄 
Thanks a lot, sounds great!

I'll have a look at it asap.

File: lib/react-archer.js [hunk 1]
```diff
@@ -0,0 +1 @@
...!0}}),e&&P(t,e)}(e,o.a.Component),function(t,e,r){e&&C(t.prototype,e),r&&C(t,r)}(e,[{key:"componentDidUpdate",value:function(t){a()(t.relations,this.props.relations)||this.registerTransitions(this.props.relations)}},{key:"componentDidMount",value:function(){0!==this.props.relations.length&&this.registerTransitions(this.props.relations)}},{key:"componentWillUnmount",value:function(){this.unregisterChild(),this.unregisterTransitions()}},{key:"render",value:function(){return o.a.createElement("div",{style:function(t){for(var e=1;e<arguments.length;e++){var r=null!=arguments[e]?arguments[e]:{},n=Object.keys(r);"function"==typeof Object.getOwnPropertySymbols&&(n=n.concat(Object.getOwnPropertySymbols(r).filter(function(t){return Object.getOwnPropertyDescriptor(r,t).enumerable}))),n.forEach(function(e){R(t,e,r[e])})}return t}({},this.props.style,{position:"relative"}),className:this.props.className,ref:this.onRefUpdate},this.props.children)}}]),e}();R(W,"defaultProps",{relations:[]});var L=function(t){return o.a.createElement(x,null,function(e){return o.a.createElement(W,A({},t,{context:e}))})};r.d(e,"ArcherElement",function(){return L}),r.d(e,"ArcherContainer",function(){return k})}])});
```
Comment (pierpo) [inline lib/react-archer.js]: Did you intend to commit this built file?

File: README.md [hunk 2]
```diff
@@ -29,8 +29,9 @@ const App = () => {
           <ArcherElement
             id="root"
             relations={[{
-              from: { anchor: 'bottom' },
-              to: { anchor: 'top', id: 'element2' }
+              id: 'element2',
```
Comment (pierpo) [inline README.md]: I'd rather have `targetId`. It's obvious but still makes it more readable at first glance.

File: lib/react-archer.js (hunk 1 above)
Comment (jfo84) [reply lib/react-archer.js]: No that's my bad. This was much easier to test with as opposed to a private npm publish flow

File: README.md (hunk 2 above)
Comment (jfo84) [reply README.md]: I like it better as well :)

**Comment by jfo84** [comment]
@pierpo I'll take a look at the bug and let you know

**Comment by jfo84** [comment]
@pierpo I fixed the bug by using a map internally. No more filtering/munging. We instead just retrieve the `sourceToTargets` by the element ID

I thought `react-fast-compare` might start tripping up with huge objects, but it appears to be quite stable

**Comment by pierpo** [comment]
I'll update the docs and publish later 😊

**Comment by jfo84** [comment]
Thank you! I'm glad I could contribute

**Comment by jfo84** [comment]
Let me know when you get around to cutting the 1.0.0 release. I can help out if needed @pierpo

**Comment by sag1v** [comment]
Ok then! i just wasted 2 hours to get it working but only now realized the API is changed lol.

Anyway, i think you should update the `README` for the repo as well :smiling_imp:

**Comment by pierpo** [comment]
@sag1v Argh, sorry about this! The README is updated but the newest version is not published so indeed new people will get the new docs while using the older version...

I'll publish right away.

**Comment by pierpo** [comment]
Should be published now. Thanks a lot @jfo84 for the huge improvements.

Sorry again @sag1v for the waste of time 😕

**Comment by pierpo** [comment]
Haha thanks for the support 😊

**Pull request merged**